### PR TITLE
remove function body from "too many args" span

### PR DESCRIPTION
**Pull request by euclio**
changelog: Remove the function body from the "too many arguments" span.

### CODE REVIEW

File: clippy_lints/src/functions.rs [hunk 1]
```diff
@@ -162,6 +162,20 @@ impl<'a, 'tcx> LateLintPass<'a, 'tcx> for Functions {
 
 impl<'a, 'tcx> Functions {
     fn check_arg_number(self, cx: &LateContext<'_, '_>, decl: &hir::FnDecl, span: Span) {
+        // Remove the function body from the span. We can't use `SourceMap::def_span` because the
+        // argument list might span multiple lines.
+        let span = match cx.sess().source_map().span_to_snippet(span) {
+            Ok(snippet) => {
+                let snippet = snippet.split('{').nth(0).unwrap_or("").trim_end();
+                if !snippet.is_empty() {
```
Comment (matthiaskrgr) [inline clippy_lints/src/functions.rs]: clippy complained about this code:
````
   Compiling clippy_lints v0.0.212 (/Users/travis/build/rust-lang/rust-clippy/clippy_lints)
error: Unnecessary boolean `not` operation
   --> clippy_lints/src/functions.rs:170:17
    |
170 | /                 if !snippet.is_empty() {
171 | |                     span.with_hi(BytePos(span.lo().0 + snippet.len() as u32))
172 | |                 } else {
173 | |                     span
174 | |                 }
    | |_________________^
    |
    = note: `-D clippy::if-not-else` implied by `-D clippy::pedantic`
    = help: remove the `!` and swap the blocks of the if/else
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#if_not_else
error: aborting due to previous error
````

File: clippy_lints/src/functions.rs [hunk 2]
```diff
@@ -162,6 +162,20 @@ impl<'a, 'tcx> LateLintPass<'a, 'tcx> for Functions {
 
 impl<'a, 'tcx> Functions {
     fn check_arg_number(self, cx: &LateContext<'_, '_>, decl: &hir::FnDecl, span: Span) {
+        // Remove the function body from the span. We can't use `SourceMap::def_span` because the
+        // argument list might span multiple lines.
+        let span = match cx.sess().source_map().span_to_snippet(span) {
```
Comment (flip1995) [inline clippy_lints/src/functions.rs]: You can use `utils::snippet_opt` here.

Comment (euclio) [reply clippy_lints/src/functions.rs]: Fixed.

File: clippy_lints/src/functions.rs (hunk 1 above)
Comment (euclio) [reply clippy_lints/src/functions.rs]: Fixed.

**Comment by flip1995** [comment]
Thanks!

@bors r+

**Pull request merged**